libethdrivers/src/plat/imx6/imx6.c

```diff
@@ -636,6 +643,18 @@ static int init_device(imx6_eth_driver_t *dev, const nic_config_t *nic_config)
 
     int ret;
 
+#if defined(CONFIG_PLAT_IMX8MQ_EVK)
+    /* Don't attempt to obtain the MAC address for iMX8MQ.
+     *
+     * The code to obtain the MAC address assumes an iMX6 compatible
+     * OCOTP which the iMX8 does not have.
+     *
+     * Instead, we assume that the bootloader has already configured the
+     * hardware MAC address. */
+    uint64_t mac = 0;
+    ZF_LOGI("using MAC configured by bootloader");
+#else
```

Comment:
I wonder if we should make this a conditional else, so we handle all known platform explicitly and don't have a default behavior? Ir would fail compilation for unknown platforms then and requires explicit action.